mono/mini/mini-ppc.c

```diff
@@ -3650,13 +3650,14 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			break;
 		case OP_GOT_ENTRY:
 			// FIXME: Fix max instruction length
-			mono_add_patch_info (cfg, offset, (MonoJumpInfoType)ins->inst_right->inst_i1, ins->inst_right->inst_p0);
+			/* XXX: This is hairy; we're casting a pointer from a union to an enum... */
```

Comment:
Is the comment correct? A union value? If so sounds like a gcc extension.

Reply:
If you follow MonoInst and the macros used, it ends up pointing to a union, and its first member. Pretty confusing, since I'm pretty sure that was the cause of some endian issues before.

Reply:
Pointing to a union and being a union are different. You can cast pointers to unions/structs, but not values, in the standard language.